### DISCUSSION TITLE
AAU, I want to have undo/redo on my tables

### DISCUSSION THREAD

**Discussion by lucasbordeau** [twentyhq/twenty#4110] [Ideas]
It can be implemented by wrapping entity data into a recoil atom containing a versioned array

**Comment by AdityaPimpalkar** [comment]
Interesting @lucasbordeau! 

Asking a few questions to understand the scope of this:
- Is it to be applied only to table cell data? (Like reverting a change made in a text cell, reverting assigned company to a person etc.)
- Should it also include keeping track of deleted rows, moved columns, hidden columns too?
- How many changes made should this versioned stack array contain? (length of versioned array)
- Should this be saved even if the user switches tables? (eg. if switched from person table to company, should it clear the version array or save it in the local state)

Thanks! :)
